storage/tianmu/core/dimension_group.cpp

```diff
@@ -91,13 +91,16 @@ DimensionGroupMaterialized::DimensionGroupMaterialized(DimensionVector &dims) {
 DimensionGroup *DimensionGroupMaterialized::Clone(bool shallow) {
   DimensionGroupMaterialized *new_value = new DimensionGroupMaterialized(dims_used);
   new_value->no_obj = no_obj;
-  if (shallow) return new_value;
   for (int i = 0; i < no_dims; i++) {
     if (t[i]) {
       new_value->nulls_possible[i] = nulls_possible[i];
-      t[i]->Lock();
-      new_value->t[i] = new IndexTable(*t[i]);
-      t[i]->Unlock();
+      if (shallow) {
+        new_value->t[i] = t[i];
+      } else {
+        t[i]->Lock();
```

Comment:
In which we delete these `new` objs.

Reply:
How about the virtual dimensionGroup

Reply:
In the destructor

```cpp

DimensionGroupMultiMaterialized::~DimensionGroupMultiMaterialized() {
  if (is_shallow_memory) {
    return;
  }

  for (auto it : dim_tables_) delete it;
}


```

Reply:
> How about the virtual dimensionGroup

DimensionGroupVirtual Filter, logic, unlike DimensionGroupMultiMaterialized, without thoroughly DimensionGroupMultiMaterialized before the whole logic, Is not recommended to modify DimensionGroupMultiMaterialized clone